### PR TITLE
Initial Implementation of Task 4 and 5

### LINKED ISSUE #9: [Task 4 + 5] Implement User Sign-Up Endpoint API
## User story
As a user, I'd like to create an account with a unique email, secure password, and a chosen username.

## Description
Implement an endpoint that allows users to register for a single session without storing any data permanently. Upon registration, each user should receive a unique session token that can be used to access protected endpoints within the session's lifetime. User data will be stored temporarily in memory and will be cleared upon application restart.

## Acceptance Criteria
- [x] POST /api/register endpoint successfully registers a new user for the session and returns a unique session token with a 201 Created response on success.
- [x] Returns a 400 Bad Request response if the email is already in use or if validation fails.
- [x] Users can log out, which invalidates their session token.
- [x] Unit tests cover email uniqueness checks, password hashing, session token generation, and access validation for protected endpoints.

### PR DESCRIPTION
**Pull request by RCruz02**
# Setup Instructions

- `npm install` to install/update all dependencies
- `npm test generateToken`
- `npm test verifyToken`
- `npm test userSignUp`

### CODE REVIEW

File: backend/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+JWT_SECRET=f7c9a8d5a13db5f9eb5f4f2872c9a158f0a6bc6fe3bf2d45c0c1a52f1b6c8bc
```
Comment (truongd3) [inline backend/.env]: We need to add file `.env` to `.gitignore`. The key should not be public.

File: backend/jwtHelper.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require('dotenv').config({ path: './backend/.env' });
+//require('dotenv').config();  // Make sure to load .env for the test environment
+
+const jwt = require('jsonwebtoken');
+
+//const JWT_SECRET = process.env.JWT_SECRET;
+const JWT_SECRET = 'f7c9a8d5a13db5f9eb5f4f2872c9a158f0a6bc6fe3bf2d45c0c1a52f1b6c8bc'
```
Comment (truongd3) [inline backend/jwtHelper.js]: I will try pick the key up instead.

File: backend/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (truongd3) [inline backend/package.json]: `package.json` and `package-lock.json` should be located in the general directory.

File: backend/jwtHelper.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+if (!JWT_SECRET) {
+  throw new Error("JWT_SECRET is not defined. Please set it in the .env file.");
+}
+
+// Function to generate JWT token
+function generateToken(email) {
```
Comment (truongd3) [inline backend/jwtHelper.js]: Break down the helper functions and save them in different files

File: tests/task4&5.test.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+const request = require('supertest');
```
Comment (truongd3) [inline tests/task4&5.test.js]: Let's rename the file

**Review by truongd3 — CHANGES_REQUESTED** [review]

**Review by RCruz02 — COMMENTED** [review]
lgtm

**Review by truongd3 — APPROVED** [review]
lgtm

**Pull request merged**